django/db/models/base.py

```diff
@@ -836,7 +836,11 @@ def delete(self, using=None):
 
         collector = Collector(using=using)
         collector.collect([self])
-        collector.delete()
+        deleted, _rows_count = collector.delete()
+        if rows_count:
```

Comment:
rows_count was proposed by akaariai (core developer) here : https://code.djangoproject.com/ticket/16891 . I can agree that single item or tuple as a return value is not a good style. But at the same time:
```
In [2]: deleted = Account.objects.first().delete()
In [3]: print(deleted)
75
```
1 account + 74 related objects  - rather confusing.
Also I'm not sure about object naming in rows_count dict keys. In most cases 'app_name.ObjName' can be loaded by apps.get_model() but in cases when m2m relation is deleted we can get something like 'queries.Item_tags' object which doesn't have associated model and can't be loaded with get_model()....